### PR TITLE
Add cpio handler

### PR DESCRIPTION
**Pull request by bmwiedemann**
it requires Archive::Cpio to do something useful,
but will not add it as a hard requirement

### CODE REVIEW

File: lib/File/StripNondeterminism/handlers/cpio.pm [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+use warnings;
+
+sub normalize {
+	my ($file) = @_;
+	eval {require Archive::Cpio} or return 0;
+	my $cpio = Archive::Cpio->new;
+	eval {$cpio->read($file)};
+	return 0 if $@; # not a cpio archive if it throws an error
+	foreach my $e ($cpio->get_files()) {
+		$e->{mtime}=0;
```
Comment (lamby) [inline lib/File/StripNondeterminism/handlers/cpio.pm]: Should this not be ``SOURCE_DATE_EPOCH`` rather than ``0``?

Comment (lamby) [inline lib/File/StripNondeterminism/handlers/cpio.pm]: Also, ``=0;`` → ``= 0;``

File: lib/File/StripNondeterminism/handlers/cpio.pm [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+#
+
+package File::StripNondeterminism::handlers::cpio;
+
+use strict;
+use warnings;
+
+sub normalize {
+	my ($file) = @_;
+	eval {require Archive::Cpio} or return 0;
```
Comment (lamby) [inline lib/File/StripNondeterminism/handlers/cpio.pm]: Please add a comment for this

Comment (lamby) [inline lib/File/StripNondeterminism/handlers/cpio.pm]: Please also specify (somewhere) what the Depends/Recommends should be for the Debian package.

**Comment by bmwiedemann** [comment]
We did not use `SOURCE_DATE_EPOCH` anywhere else, so I went with canonical_time which was used in other places.

**Comment by bmwiedemann** [comment]
and I added some lines about debian packages to the first commit message - did not find a good file to add it to

**Comment by lamby** [comment]
> We did not use SOURCE_DATE_EPOCH anywhere else, so I went with canonical_time which was used in other places.

My bad; yeah, using ``canonical_time`` is correct (it's populated from ``SOURCE_DATE_EPOCH`` hence why I first said that :))

> and I added some lines about debian packages to the first commit message - did not find a good file to add it to

Thanks! Except ``E: Unable to locate package libarchive-cpio-perl`` :) What am I missing?

**Comment by lamby** [comment]
@bmwiedemann ^

**Comment by bmwiedemann** [comment]
seems to be a missing package from CPAN. That is why I omitted the tests in the first version, because those would require it

**Comment by lamby** [comment]
> seems to be a missing package from CPAN. That is why I omitted the tests in the first version, because those would require it

Sure, but the testsuite will now always fail on Debian…

**Comment by bmwiedemann** [comment]
filed https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=861948 for libarchive-cpio-perl in Debian

**Comment by lamby** [comment]
@bmwiedemann Thanks!

**Comment by bmwiedemann** [comment]
re-added the 2nd commit with the cpio tests, now that we have a Debian package for it.

**Comment by lamby** [comment]
Great stuff. I'll merge this into an ``experimental`` branch; I want to leave the main branch clear in case we need to do another upload targetting ``stretch``.

**Comment by lamby** [comment]
Merged to ``experimental``

**Pull request closed without merging**